tests/cases/fourslash/extract-method14.ts

```diff
@@ -0,0 +1,24 @@
+/// <reference path='fourslash.ts' />
+
+// Don't emit type annotations in JavaScript files
+// Also tests that single-variable return extractions don't get superfluous destructuring
+
+// @allowNonTsExtensions: true
+// @Filename: foo.js
+//// function foo() {
+////     var i = 10;
+////     /*a*/return i++;/*b*/
+//// }
+
+goTo.select('a', 'b');
+edit.applyRefactor('Extract Method', 'scope_1');
+verify.currentFileContentIs(`function foo() {
+    var i = 10;
+    var __return: any;
+    ({ i, __return } = newFunction(i));
+    return __return;
+}
+function newFunction(i) {
+    return { i, __return: i++ };

```

Comment:
You assign `i` before `i++` happens. Meaning at the call site, `i` will not actually have the incremented value. In this case, that's fine because you return immediately, but here's a case that breaks:


```ts
function foo() {
    var i = 10;
    
    function inner() {
        return i++;  // extract this line into file scope, the console output changes
    }

    inner();
    console.log(i);
} 

foo();
```